### PR TITLE
[formrecognizer] refactor helpers out of models

### PR DESCRIPTION
**Pull request by kristapratico**
This PR does a few things...

- Refactors some of the helper functions out of the models file
- Refactors how element pointers are resolved so that it is more accepting of adding new types of elements
    - avoids a potential bug that could occur when new types introduced
    - adds tests for this
- corrects some formatting/typos

Live test pass: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=484667&view=results

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_resolve_elements.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    def test_word_reference(self):
+        # unlabeled
+        word_unlabeled_ref = "#/readResults/13/lines/91/words/1000"
+        assert get_element_type(word_unlabeled_ref) == "word"
+        indices = [int(s) for s in re.findall(r"\d+", word_unlabeled_ref)]
+        assert [13, 91, 1000] == indices
+
+        # labeled
+        word_labeled_ref = "#/analyzeResult/readResults/0/lines/46/words/0"
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/tests/test_resolve_elements.py:23]: can you also add a test where there's a number after `analyzeResult`

Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_resolve_elements.py:23]: discussed offline

**Pull request merged**